resources/views/components/game-progress-bar.blade.php

```diff
@@ -0,0 +1,75 @@
+@props([
+    'awardIndicator' => null, // 'unfinished' | 'beaten-softcore' | 'beaten-hardcore' | 'completed' | 'mastered' | null
+    'softcoreProgress' => 0,
+    'hardcoreProgress' => 0,
+    'maxProgress' => 100,
+    'tooltip' => null, // ?string
+])
+
+<?php
+    $hardcoreProgressBarWidth = $softcoreProgressBarWidth = '0';
+
+    if ($maxProgress > 0) {
+        if ($softcoreProgress >= $maxProgress) {
+            $softcoreProgressBarWidth = sprintf("%01.2f", ($maxProgress - $hardcoreProgress) * 100 / $maxProgress);
+        } elseif ($softcoreProgress > 0) {
+            $softcoreProgressBarWidth = sprintf("%01.2f", ($softcoreProgress - $hardcoreProgress) * 100 / $maxProgress);
+        }
+
+        if ($hardcoreProgress >= $maxProgress) {
+            $hardcoreProgressBarWidth = '100.0';
+        } elseif ($hardcoreProgress > 0) {
+            $hardcoreProgressBarWidth = sprintf("%01.2f", $hardcoreProgress * 100 / $maxProgress);
+        }
+    }
+
+    if (!$tooltip) {
+        $tooltip = "{$softcoreProgress} ({$hardcoreProgress}) of {$maxProgress}";
+    }
+?>
+
+<div class="relative">
+    <div
+        role="progressbar"
+        aria-valuemin="0"
+        aria-valuemax="{{ $maxProgress }}"
+        aria-valuenow="{{ $softcoreProgress }}"
+        aria-label="{{ $tooltip }}"
+        title="{{ $tooltip }}"
+        class="w-full h-1 bg-zinc-950 light:bg-zinc-300 rounded flex space-x-px overflow-hidden"
+    >
+        @if ($hardcoreProgress > 0)
+            <div
+                style="width: {{ $hardcoreProgressBarWidth }}%"
+                class="bg-gradient-to-r from-amber-500 to-[gold] light:bg-yellow-500 h-full"
+            ></div>
+        @endif
+
+        @if ($softcoreProgress > $hardcoreProgress)
+            <div
+                style="width: {{ $softcoreProgressBarWidth }}%"
+                class="bg-neutral-500 h-full"
+            ></div>
+        @endif
+    </div>
+
+    @if ($awardIndicator)
+        @php
+            $awardTitles = [
+                'unfinished' => 'Unfinished',
+                'beaten-softcore' => 'Beaten (softcore)',
+                'beaten-hardcore' => 'Beaten',
+                'completed' => 'Completed',
+                'mastered' => 'Mastered',
+            ];
+        @endphp
+
+        <div
+            class="gprogress-ind__root"
+            data-award="{{ $awardIndicator }}"
+            title="{{ $awardTitles[$awardIndicator] ?? 'Unfinished' }}"
```

Comment:
I'm not sure I like seeing the items as mastered for revised items.

Before:
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/ba65dd65-b9c0-430f-b897-8049c991a600)
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/ea43b4a7-1f9c-481a-8adf-1819e2ad2a8e)

After:
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/5a237941-b5f0-4efc-a058-b30d885d0d92)
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/fb4164b9-0ca5-4acc-91cd-a5d9288170fb)

In particular, the second one looks like it has been fully mastered because the mastery circle obscures the incomplete bar.

Reply:
Retired sets too:
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/1fbb6bf9-fdd2-4904-982f-2d3dbab090a9)

Reply:
Two changes made in latest:
* Sets with 0 achievements earned (retired sets) are now hidden from the list.
* Some additional logic has been added to not show masteries/completions if the user hasn't earned 100% of the achievements in hardcore/softcore for the set. Now, the row will check if a beaten award was earned and fall back to that, otherwise use the "unfinished" designator.

Reply:
Verified that retired sets are no longer visible, and while Super Mario 64 and Shantae do show as Beaten now, I still have some revisions appearing as Mastered:
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/ff38a1da-7380-438a-ba08-39eed0504a50)

![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/98704649-b175-4676-813c-d95d2c3906c5)
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/a7468178-ef58-4ed2-afb9-93a10e7596ed)
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/8de5680e-cb9d-4574-959e-59eb65c064e8)

Reply:
Found the logic error - addressed in latest.

Reply:
Crystalis and DEcapAttack should be showing as beaten, but now they show as incomplete. Jaws is also showing as incomplete, but the revision included a progression achievement.

Reply:
Found a separate logic error in `PlayerProgressionService`. The beaten awards weren't being persisted correctly in `$allAwardsByGameId`.